tests/test_datadog.py

```diff
@@ -42,3 +44,55 @@ def test_api_key():
     configured_api_key = DataDog.settings[DataDog.KEY_DATADOG_API_KEY]
 
     assert configured_api_key == 'test_api_key'
+
+
+def test_metrics():
```

Comment:
Sorry about that. This wasn't meant to come across as a "do it the way I like it suggestion", I'll try and phase it more appropriately next time. 

I was merely trying to suggest that there might be some value in better failure reporting when using separate functions compared to the existing way. And I totally agree that the duplicated setup effort is a downside. Maybe a pytest `fixture` would be helpful to avoid that?